packages/ckeditor5-ui/src/tooltipmanager.ts

```diff
@@ -467,7 +467,21 @@ export default class TooltipManager extends DomEmitterMixin() {
 	 * Hides the tooltip when the element is no longer visible in DOM or the tooltip text was removed.
 	 */
 	private _updateTooltipPosition() {
-		const tooltipData = getTooltipData( this._currentElementWithTooltip! );
+		// Actually, the #_currentElementWithTooltip member can occasionally be null. This can happen in the following scenario:
+		//
+		// 1. The user opens the balloon editor in a content editable.
+		// 2. The user clicks the "Remove link" button, which is attached to the balloon and the link in the editor.
+		// 3. The editor removes the link (and the attached balloon) from the editable.
+		// 4. The unmounted balloon fires an `update` event (for example, from form `blur` event).
+		// 5. We receive the `update` event, and #_currentElementWithTooltip is null.
+		// 6. Ballon fully unmounts and unbinds all `blur` listeners.
```

Comment:
I think those steps do not describe the real issue and are incorrect.

> 1. The user opens the balloon editor in a content editable.

It doesn't have to be a balloon editor, so this is an incorrect statement. Perhaps it was meant to refer to a balloon panel. In the next point, there is a reference to "Remove link", so it might be worth adding information that the example is a link balloon.

> 2. The user clicks the "Remove link" button, which is attached to the balloon and the link in the editor.

In reproducing the issue, it is crucial that the tooltip is attached to the button. This information is missing here, and it is found at the end of the comment.

> 3. The editor removes the link (and the attached balloon) from the editable.

This is also unclear - there is no information about what balloon content actually is. In the issue I tested, it specifically concerns a button in the balloon.


I'm attaching the video with the reproduction steps below:

https://github.com/ckeditor/ckeditor5/assets/20561161/83128dd5-2c03-4333-b45e-bf023cedb871